trade-finance-manager-ui/scripts/find-utilisation-reports-by-year-year-input-dropdown.ts

```diff
@@ -0,0 +1,29 @@
+const setupFindUtilisationReportsByYearYearInput = (): void => {
+  const findUtilisationReportsByYearForm: HTMLFormElement | null = document.querySelector('form#find-utilisation-reports-by-year--form');
+
+  if (!findUtilisationReportsByYearForm) {
+    return;
+  }
+
+  const yearInput: HTMLInputElement | null = document.querySelector('input#find-utilisation-reports-by-year--year-input');
+  if (!yearInput) {
+    return;
+  }
+
+  const bankRadios: NodeListOf<HTMLInputElement> = findUtilisationReportsByYearForm.querySelectorAll('input.govuk-radios__input');
+
+  bankRadios.forEach((bankRadio) => {
+    const bankId = bankRadio.value;
+    const dataListId = `datalist--bankId-${bankId}`;
+
+    if (bankRadio.checked) {
+      yearInput.setAttribute('list', dataListId);
+    }
+
+    bankRadio.addEventListener('click', () => {
+      yearInput.setAttribute('list', dataListId);
+    });
+  });
+};
+
+setupFindUtilisationReportsByYearYearInput();
```

Comment:
Why do we need a client side script to set some attributes, can this not be done in nunjucks?

If not, will the core functionality continue to work when JavaScript is disabled?

Reply:
This cannot be done in nunjucks - it is something that updates attributes based on user input. The page is already designed to work without JS being enabled - this is an add-on which was requested to get some nicer behaviour when JS is enabled

Reply:
Designed to work without JS being enabled? 😞

Reply:
Yes - you can always input a year without JS being enabled. This script simply means that you will have a helpful dropdown that appears displaying the years where we know reports exist when you are typing a value into the year input